File: pi-deploy/wilabia-backend/inject-aegis-middleware.js
#!/usr/bin/env node
/**
 * Idempotent patcher for ~/wilabia-original/backend/src/index.js
 * Inserts the AEGIS unified-feed middleware after the CORS block and before
 * the rate-limiter chain (per recon report — file is at line 152 end of cors).
 */
const fs = require('fs');
const TARGET = '/Users/alejandxr/wilabia-original/backend/src/index.js';

const src = fs.readFileSync(TARGET, 'utf8');
if (src.includes('AEGIS unified-feed middleware')) {
  console.log('AEGIS middleware already injected — skipping');
  process.exit(0);
}

// Match the anchor with flexible whitespace around the closing }));
const ANCHOR_RE = /allowedHeaders: \['Content-Type', 'Authorization', 'X-Filename', 'X-Push-Endpoint', 'Cache-Control', 'Pragma'\]\r?\n\}\)\);\r?\n/;
const m = src.match(ANCHOR_RE);
if (!m) {
  console.error('ERROR: anchor not found in target file (cors block end)');
  process.exit(1);
}
const ANCHOR = m[0];

const BLOCK = `
// === AEGIS unified-feed middleware ===
// Emits one JSON line per request to /Users/alejandxr/web-logs/aegis-feed.jsonl
// which AEGIS log_watcher tails. trust proxy=1 is already set on line 71 so
// req.ip resolves the real client IP from cf-connecting-ip / x-forwarded-for.
// Listens on res.on('finish') so the response status is correct.
app.use((req, res, next) => {
  res.on('finish', () => {
    try {
      const ip = req.headers['cf-connecting-ip']
                 || (req.headers['x-forwarded-for'] || '').split(',')[0].trim()
                 || req.ip;
      const record = {
        ts: new Date().toISOString(),
        app: 'wilabia-backend',
        src_ip: ip,
        method: req.method,
        path: req.originalUrl,
        status: res.statusCode,
      };
      const ua = (req.headers['user-agent'] || '').substring(0, 300);
      const host = req.headers['host'] || '';
      const ref = req.headers['referer'] || '';
      const fwd = req.headers['x-forwarded-for'] || '';
      const cfray = req.headers['cf-ray'] || '';
      const country = req.headers['cf-ipcountry'] || '';
      if (ua) record.ua = ua;
      if (host) record.host = host;
      if (ref) record.ref = ref;
      if (fwd) record.fwd_chain = fwd;
      if (cfray) record.cf_ray = cfray;
      if (country) record.country = country;
      fs.appendFileSync('/Users/alejandxr/web-logs/aegis-feed.jsonl', JSON.stringify(record) + '\\n');
    } catch (_) { /* never let logging take down the request */ }
  });
  next();
});

`;

const out = src.replace(ANCHOR, ANCHOR + BLOCK);
fs.writeFileSync(TARGET, out);
console.log(`OK: injected ${BLOCK.length} bytes`);

File: pi-deploy/sid-backend/inject-aegis-middleware.py
#!/usr/bin/env python3
"""Idempotent patcher for sid-backend FastAPI main.py.
Adds AEGIS access-log middleware after CORSMiddleware.
"""
import sys

TARGET = "/Users/alejandxr/sid-wilab/backend/main.py"

with open(TARGET) as f:
    src = f.read()

if "AEGIS unified-feed middleware" in src:
    print("AEGIS middleware already injected — skipping")
    sys.exit(0)

ANCHOR = '''app.add_middleware(
    CORSMiddleware,
    allow_origins=ALLOWED_ORIGINS,
    allow_credentials=True,
    allow_methods=["GET", "POST", "PUT", "DELETE", "OPTIONS"],
    allow_headers=["*"],
)
'''

INJECT = '''
# === AEGIS unified-feed middleware ===
# Emits one JSON line per HTTP request to /Users/alejandxr/web-logs/aegis-feed.jsonl
# so AEGIS log_watcher gets HTTP-layer visibility with real client IPs.
# Requires uvicorn to be launched with --proxy-headers --forwarded-allow-ips=127.0.0.1
# AND for the upstream proxy (web-server-logger.py) to inject X-Forwarded-For.
import json as _aegis_json
import time as _aegis_time
from datetime import datetime as _aegis_dt, timezone as _aegis_tz
_AEGIS_FEED = os.environ.get(
    "AEGIS_FEED_PATH", "/Users/alejandxr/web-logs/aegis-feed.jsonl"
)

@app.middleware("http")
async def _aegis_access_log(request, call_next):
    _start = _aegis_time.time()
    response = await call_next(request)
    try:
        _hdr = request.headers
        _src_ip = (
            _hdr.get("cf-connecting-ip")
            or (_hdr.get("x-forwarded-for") or "").split(",")[0].strip()
            or _hdr.get("x-real-ip")
            or (request.client.host if request.client else "unknown")
        )
        _now = _aegis_dt.now(_aegis_tz.utc)
        _record = {
            "ts": _now.strftime("%Y-%m-%dT%H:%M:%S.") + f"{_now.microsecond // 1000:03d}Z",
            "app": "sid-backend",
            "src_ip": _src_ip,
            "method": request.method,
            "path": str(request.url.path) + ("?" + request.url.query if request.url.query else ""),
            "status": response.status_code,
            "rt_ms": int((_aegis_time.time() - _start) * 1000),
        }
        for _k, _hk in (
            ("ua", "user-agent"),
            ("ref", "referer"),
            ("host", "host"),
            ("country", "cf-ipcountry"),
            ("fwd_chain", "x-forwarded-for"),
            ("cf_ray", "cf-ray"),
        ):
            _v = _hdr.get(_hk)
            if _v:
                _record[_k] = _v
        with open(_AEGIS_FEED, "a", buffering=1) as _fh:
            _fh.write(_aegis_json.dumps(_record, ensure_ascii=False) + "\\n")
    except Exception:
        pass  # never let logging break the request
    return response

'''

if ANCHOR not in src:
    print("ERROR: CORSMiddleware anchor not found", file=sys.stderr)
    sys.exit(1)

new_src = src.replace(ANCHOR, ANCHOR + INJECT, 1)
with open(TARGET, "w") as f:
    f.write(new_src)
print(f"OK: injected {len(INJECT)} bytes")
